artiq/gui/log.py

```diff
@@ -168,6 +191,8 @@ def data(self, index, role):
             return (log_level_to_name(v[0]) + ", " +
                 time.strftime("%m/%d %H:%M:%S", time.localtime(v[2])) +
                 "\n" + v[3][lineno])
+        elif role == None:
```

Comment:
Why not pass a valid role in ``filterAcceptsRow()`` instead?

Reply:
The `DisplayRole` is used in line 172 to get the message of each row and it will not be shown if its parents does not fit the text filter. Therefore it has to return the original entry for checking.

Reply:
I don't understand. Where does ``_Model`` know about that filter?

Reply:
`_ProxyModel` is acting as the filter and `_Model` just responsible for passing the data

Reply:
Exactly. So just pass a valid role to ``_Model`` in ``filterAcceptsRow`` instead of abusing Python typing. What is the problem?